### PR TITLE
PYTHON-4701 - Topology logging should use suppress_event

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: pymongo/asynchronous/topology.py [hunk 1]
```diff
@@ -497,7 +497,7 @@ async def _process_change(
                     (td_old, self._description, self._topology_id),
                 )
             )
-        if _SDAM_LOGGER.isEnabledFor(logging.DEBUG):
+        if _SDAM_LOGGER.isEnabledFor(logging.DEBUG) and sd_old != server_description:
```
Comment (ShaneHarvey) [inline pymongo/asynchronous/topology.py:500]: Should this be ` and not suppress_event`?

**Comment by ShaneHarvey** [comment]
Also could you add some more info to the failing assert message?

File: pymongo/asynchronous/topology.py (hunk 1 above)
Comment (NoahStapp) [reply pymongo/asynchronous/topology.py:500]: `suppress_event` checks for event publishing flags, which we don't use for logging.

**Comment by NoahStapp** [comment]
> Also could you add some more info to the failing assert message?

To which message?

File: pymongo/asynchronous/topology.py (hunk 1 above)
Comment (ShaneHarvey) [reply pymongo/asynchronous/topology.py:500]: Could that be changed?
```
suppress_event = sd_old == server_description
```

**Review by ShaneHarvey — APPROVED** [review]

**Comment by ShaneHarvey** [comment]
Could you update the PR title to describe the bug fix instead of the test failure name?

**Pull request merged**